src/github.com/getlantern/flashlight/app/settings.go

```diff
@@ -145,6 +145,14 @@ func (s *Settings) checkBool(data map[string]interface{}, name string, f func(bo
 	}
 }
 
+func (s *Settings) checkInt(data map[string]interface{}, name string, f func(int)) {
+	if v, ok := data[name].(int); ok {
```

Comment:
Does that casting actually work if we send a number there? I can't check
right now but, as far as I know, since data is a map of interface{} values
the value will be unmarshaled into a float64, and we need to convert that
to int.
On Thu, May 19, 2016 at 20:56 myleshorton <notifications@github.com> wrote:

> In src/github.com/getlantern/flashlight/app/settings.go
> <https://github.com/getlantern/lantern/pull/4345#discussion_r63981775>:
>
> > @@ -145,6 +145,14 @@ func (s *Settings) checkBool(data map[string]interface{}, name string, f func(bo
> >  	}
> >  }
> >
> > +func (s *Settings) checkInt(data map[string]interface{}, name string, f func(int)) {
> > +	if v, ok := data[name].(int); ok {
>
> I switched this to cast it to an int @xiam <https://github.com/xiam>, as
> using float64 made the tests fail. What was the thought behind float64
> versus int?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/getlantern/lantern/pull/4345/files/17346b9c1bbc898f0a06aeeb0bd168ed5fe363c6#r63981775>
>